src/fides/api/service/connectors/rds_postgres_connector.py

```diff
@@ -0,0 +1,152 @@
+        """
+        Pre client hook for RDS Postgres Connector
+        """
+        db: Session = Session.object_session(self.configuration)
+        self.namespace_meta = SQLConnector.get_namespace_meta(db, node.address.dataset)
+
+    # Overrides SQLConnector.create_client
+    def create_client(self) -> Engine:
+        """
+        Returns a SQLAlchemy Engine that can be used to interact with a database
+        """
+        if self.namespace_meta is None:
+            raise ConnectionException(
+                "Namespace meta is not set. Please call pre_client_creation_hook before creating the client."
+            )
+
+        database_instance_name = self.namespace_meta["database_instance_id"]
+        db_name = self.namespace_meta["database_id"]
+        db_info = self.get_database_instance_connection_info(database_instance_name)
+        return self.create_engine(
+            db_username=self.typed_secrets.db_username,
+            host=db_info["host"],
+            port=db_info["port"],
+            db_name=db_name,
+        )
+
+    def query_config(self, node: ExecutionNode) -> SQLQueryConfig:
+        """Query wrapper corresponding to the input execution_node."""
+        return PostgresQueryConfig(node)
+
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        """
+        Connects to AWS Account tries to describe RDS instances and clusters.
+        """
+        logger.info("Starting test connection to {}", self.configuration.key)
+
+        try:
+            rds_client = self.rds_client
+        except ClientError as e:
+            raise ConnectionException(
+                f"Error creating RDS client {self.configuration.key}: {e}"
+            )
+
+        try:
+            logger.info("Describing RDS clusters for {}", self.configuration.key)
+            rds_client.describe_db_clusters()
+        except ClientError as e:
+            raise ConnectionException(
+                f"Error describing RDS clusters {self.configuration.key}: {e}"
+            )
+
+        try:
+            logger.info("Describing RDS instances for {}", self.configuration.key)
+            rds_client.describe_db_instances()
+
+        except ClientError as e:
+            raise ConnectionException(
+                f"Error describing RDS instances {self.configuration.key}: {e}"
+            )
+
+        return ConnectionTestStatus.succeeded
+
+    @staticmethod
+    def cursor_result_to_rows(results: LegacyCursorResult) -> List[Row]:
+        """
+        Convert SQLAlchemy results to a list of dictionaries
+        """
+        return SQLConnector.default_cursor_result_to_rows(results)
+
+    def retrieve_data(
+        self,
+        node: ExecutionNode,
+        policy: Policy,
+        privacy_request: PrivacyRequest,
+        request_task: RequestTask,
+        input_data: Dict[str, List[Any]],
+    ) -> List[Row]:
+        """DSR execution not yet supported for RDS Postgres"""
+        return []
```

Comment:
I think we should probably raise a `NotImplemented` error here right? we don't w ant our code calling these methods and making it seem like it's actually doing things? or am I missing something?

Reply:
honestly I'm not too sure if raising NotImplemented will make it so that DSRs fail when the connector is configured -- let's keep existing behavior for now and maybe discuss this more thoroughly another time